### PR TITLE
Issue 5388: Enhancements to Integration test utility ClusterWrapper

### LINKED ISSUE #5388: Enhancements for ClusterWrapper - an Integration test server runner
**Problem description**
Here are a couple of additional enhancement needed for the `ClusterWrapper`, which can be used for running a cluster for integration testing. 

* It does not support running Controller REST API
* The cluster is started using `initialize()` method. It should be renamed to `start()`. 
* A few more examples of its usage'd be helpful for developers who are new to the class. 

**Problem location**
https://github.com/pravega/pravega/blob/22996fc928d762f886265a3dee75157fac944357/test/integration/src/main/java/io/pravega/test/integration/demo/ClusterWrapper.java

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by ravisharda**
**Change log description**  
Enhance `ClusterWrapper` to include support for Controller REST API, rename `initialize()` to `start()` and add usage examples. 

**Purpose of the change**  
Resolves #5388. 

**What the code does**  
* Adds support for enabling Controller REST interface.
* Renames `initialize()` method to `start()` .
* Add some more integration tests for demonstration in `ClusterWrapperTest`.
* Rearranges some code in the `ClusterWrapper` to collocate all public methods.

**How to verify it**  
All integration tests must pass.

### CODE REVIEW

**Review by anirudhkovuru — APPROVED** [review]
LGTM

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java [hunk 1]
```diff
@@ -94,4 +127,81 @@ public void writeAndReadBackAMessageWithTlsAndAuthEnabledServer() {
...
+        // Check if the response was as expected
+        Response response = builder.get();
+        assertEquals("Response to /ping was not OK", OK.getStatusCode(), response.getStatus());
+        log.info("Ping successful.");
+    }
+
+    //region Unit test
+    @Test
+    public void setsDefaultValuesWhenBuilderSpecifiesNoValues() {
+        ClusterWrapper objectUnderTest = ClusterWrapper.builder().build();
```
Comment (andreipaduroiu) [inline test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:196]: You should clean this up

**Review by andreipaduroiu — CHANGES_REQUESTED** [review]

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java (hunk 1 above)
Comment (ravisharda) [reply test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java]: Ok. I've removed the unit tests from this integration test class in the latest commit. Have pushed the assertions for checking default values into an integration test above.

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java [hunk 2]
```diff
@@ -94,4 +136,66 @@ public void writeAndReadBackAMessageWithTlsAndAuthEnabledServer() {
...
+
+        // Prepare a TLS context with truststore containing the signing CA's vertificate
+        KeyStore trustStore = createTrustStore(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_CA_CERT_FILE_NAME);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(trustStore);
+        SSLContext tlsContext = SSLContext.getInstance("TLS");
+        tlsContext.init(null, tmf.getTrustManagers(), null);
+
+        // Instantiate the REST client
+        javax.ws.rs.client.Client client = ClientBuilder.newBuilder()
```
Comment (shrids) [inline test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:183]: cleanup using `client.close()`

**Review by shrids — APPROVED** [review]
LGTM apart from one comment.

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java (hunk 2 above)
Comment (ravisharda) [reply test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:183]: Thanks for catching it - missed it.

File: test/integration/src/main/java/io/pravega/test/integration/demo/ClusterWrapper.java [hunk 3]
```diff
@@ -122,30 +137,64 @@
...
+        }
+        startZookeeper();
+        log.info("Started Zookeeper");
+        startSegmentStore();
+        log.info("Started Segment store");
+        startController();
+        log.info("Started Controller");
+    }
+
+    @Override
```
Comment (andreipaduroiu) [inline test/integration/src/main/java/io/pravega/test/integration/demo/ClusterWrapper.java:156]: Why did you move this method? I have another PR in flight touching this class and it's going to conflict with this one because of this change

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java [hunk 4]
```diff
@@ -94,4 +136,69 @@ public void writeAndReadBackAMessageWithTlsAndAuthEnabledServer() {
...
+        org.glassfish.jersey.client.ClientConfig clientConfig = new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.register(JacksonJsonProvider.class);
+        clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
+        clientConfig.register(HttpAuthenticationFeature.basic(restApiUser, restApiUserPwd));
+
+        // Prepare a TLS context with truststore containing the signing CA's vertificate
+        KeyStore trustStore = createTrustStore(TestUtils.pathToConfig() + SecurityConfigDefaults.TLS_CA_CERT_FILE_NAME);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(trustStore);
+        SSLContext tlsContext = SSLContext.getInstance("TLS");
```
Comment (andreipaduroiu) [inline test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:179]: Do any of these need cleaning up?

File: test/integration/src/main/java/io/pravega/test/integration/demo/ClusterWrapper.java (hunk 3 above)
Comment (ravisharda) [reply test/integration/src/main/java/io/pravega/test/integration/demo/ClusterWrapper.java:156]: Since I was enhancing this class, I thought it'd be a good opportunity to move around methods so that public and private methods are collocated.

File: test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java (hunk 4 above)
Comment (ravisharda) [reply test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:179]: This is alright. It's creating a truststore containing the CA certificate, so that the server certificate can be trusted by the client.

Comment (ravisharda) [reply test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java:179]: Also, this is where we read certs from files, and we close the stream there: 

https://github.com/pravega/pravega/blob/2e70f876c5eaeaf0ed51504174c5e6b13f0645a1/common/src/main/java/io/pravega/common/util/CertificateUtils.java#L40-L47

**Review by andreipaduroiu — APPROVED** [review]
